MoltenVK/MoltenVK/GPUObjects/MVKQueue.mm

```diff
@@ -85,7 +85,16 @@
 	// The submissions will ensure a misconfiguration will be safe to execute.
 	VkResult rslt = qSubmit->getConfigurationResult();
 	if (_execQueue) {
-		dispatch_async(_execQueue, ^{ execute(qSubmit); } );
+		std::unique_lock lock(_execQueueMutex);
+		_execQueueJobCount++;
+
+		dispatch_async(_execQueue, ^{
+			execute(qSubmit);
+
+			std::unique_lock execLock(_execQueueMutex);
+			_execQueueJobCount--;
+			_execQueueConditionVariable.notify_all();
```

Comment:
The wait code below checks that the job count is zero before actually exiting the wait. You could probably avoid spurious wakeups by only notifying when `_execQueueJobCount` becomes zero. (Or did you try that and experience problems?)

Reply:
No problems, I think that's good.